course/src/isometry.cc

```diff
@@ -167,8 +167,61 @@ double Matrix3::det() const {
       rows_[0][2] * (rows_[1][0] * rows_[2][1] - rows_[1][1] * rows_[2][0]));
 }
 
+Matrix3 Matrix3::product(const Matrix3 &rhs) const {
+  Matrix3 output;
+
+  for (int i = 0; i < 3; ++i) {
+    for (int j = 0; j < 3; ++j) {
+      output[i][j] = row(i).dot(rhs.col(j));
+    }
+  }
+  return output;
+}
+
+Vector3 Matrix3::product(const Vector3 &rhs) const {
+  Vector3 output;
+  for (int i = 0; i < 3; ++i) {
+    output[i] = row(i).dot(rhs);
+  }
+  return output;
+}
+
 const Matrix3 Matrix3::kIdentity = {1., 0., 0., 0., 1., 0., 0., 0., 1.};
 const Matrix3 Matrix3::kOnes = {1., 1., 1., 1., 1., 1., 1., 1., 1.};
 const Matrix3 Matrix3::kZero = {0., 0., 0., 0., 0., 0., 0., 0., 0.};
 
+const Isometry Isometry::FromTranslation(const Vector3 &vec) {
+  return Isometry{vec, Matrix3::kIdentity};
+}
+
+Isometry Isometry::RotateAround(const Vector3 &direction, const double &value) {
+  Isometry output;
+
+  if (direction == Vector3::kUnitX) {
```

Comment:
This function should work for **any** `direction` vector. See [this](https://en.wikipedia.org/wiki/Rotation_matrix#Rotation_matrix_from_axis_and_angle). You'll have to make sure you normalize the direction vector.

Reply:
Done